src/reference/00-Getting-Started/07A-Task-Graph.md

```diff
@@ -0,0 +1,324 @@
+
+For example, if you `inspect tree compile` you'll see it depends on another key
+`incCompileSetup`, which it in turn depends on
+other keys like `dependencyClasspath`. Keep following the dependency chains and magic happens.
+
+```
+> inspect tree compile
+[info] compile:compile = Task[sbt.inc.Analysis]
+[info]   +-compile:incCompileSetup = Task[sbt.Compiler\$IncSetup]
+[info]   | +-*/*:skip = Task[Boolean]
+[info]   | +-compile:compileAnalysisFilename = Task[java.lang.String]
+[info]   | | +-*/*:crossPaths = true
+[info]   | | +-{.}/*:scalaBinaryVersion = 2.12
+[info]   | |
+[info]   | +-*/*:compilerCache = Task[xsbti.compile.GlobalsCache]
+[info]   | +-*/*:definesClass = Task[scala.Function1[java.io.File, scala.Function1[java.lang.String, Boolean]]]
+[info]   | +-compile:dependencyClasspath = Task[scala.collection.Seq[sbt.Attributed[java.io.File]]]
+[info]   | | +-compile:dependencyClasspath::streams = Task[sbt.std.TaskStreams[sbt.Init\$ScopedKey[_ <: Any]]]
+[info]   | | | +-*/*:streamsManager = Task[sbt.std.Streams[sbt.Init\$ScopedKey[_ <: Any]]]
+[info]   | | |
+[info]   | | +-compile:externalDependencyClasspath = Task[scala.collection.Seq[sbt.Attributed[java.io.File]]]
+[info]   | | | +-compile:externalDependencyClasspath::streams = Task[sbt.std.TaskStreams[sbt.Init\$ScopedKey[_ <: Any]]]
+[info]   | | | | +-*/*:streamsManager = Task[sbt.std.Streams[sbt.Init\$ScopedKey[_ <: Any]]]
+[info]   | | | |
+[info]   | | | +-compile:managedClasspath = Task[scala.collection.Seq[sbt.Attributed[java.io.File]]]
+[info]   | | | | +-compile:classpathConfiguration = Task[sbt.Configuration]
+[info]   | | | | | +-compile:configuration = compile
+[info]   | | | | | +-*/*:internalConfigurationMap = <function1>
+[info]   | | | | | +-*:update = Task[sbt.UpdateReport]
+[info]   | | | | |
+....
+```
+
+When you type `compile` sbt automatically performs an `update`, for example. It
+Just Works because the values required as inputs to the `compile`
+computation require sbt to do the `update` computation first.
+
+In this way, all build dependencies in sbt are *automatic* rather than
+explicitly declared. If you use a key's value in another computation,
+then the computation depends on that key.
+
+#### Declaring dependency to other settings
+
+Next, take these two keys (from [Keys](../sxr/sbt/Keys.scala.html)):
+
+```scala
+val scalacOptions = taskKey[Seq[String]]("Options for the Scala compiler.")
+val checksums = settingKey[Seq[String]]("The list of checksums to generate and to verify for dependencies.")
+```
+
+**Note**: `scalacOptions` and `checksums` have nothing to do with each other, they
+are just two keys with the same value type, where one is a task.
+
+It is possible to compile a `build.sbt` that aliases `scalacOptions` to
+`checksums`, but not the other way. For example, this is allowed:
+
+```scala
+// The scalacOptions task may be defined in terms of the checksums setting
+scalacOptions := {
+  val cs = checksums.value
+  cs
```

Comment:
We need to have more realistic, motivated examples for these introductory snippets. In it's current state, it's somewhat too abstract, and reading it I'm not sure what i'm trying to learn. This is especially true because we never "run" the examples, so I can't even really "see" what the result of each of these things is.

Furthermore, if we come up with realistic examples for every code snippet, apart from teaching the newbies how the *graph DSL* works, it will simultaneously teach them how to perform basic tasks, so they'd learn more useful things for the same amount of work